src/convert.js

```diff
@@ -125,7 +126,7 @@ function buf2str (buf) {
 
 function mh2buf (hash) {
   // the address is a varint prefixed multihash string representation
-  const mh = Buffer.from(bs58.decode(hash))
+  const mh = bs58.decode((new CID(hash)).toString('base58btc'))
```

Comment:
I think we could remove 1 level of string serialization and use buffer already present in CID?
```suggestion
  const mh = new CID(hash).multihash
```

Reply:
👍 that's much better, and also resolves a `toString()` issue I was seeing (and added a test for)